engine/env/spark/src/main/java/org/datacleaner/spark/Main.java

```diff
@@ -39,20 +56,53 @@ public static void main(String[] args) {
         final String confXmlPath = args[0];
         final String analysisJobXmlPath = args[1];
 
-        final String propertiesPath; 
+        final String propertiesPath;
         if (args.length > 2) {
             propertiesPath = args[2];
         } else {
             propertiesPath = null;
         }
 
-        final SparkJobContext sparkJobContext = new SparkJobContext(sparkContext, confXmlPath, analysisJobXmlPath, propertiesPath);
-
+        final SparkJobContext sparkJobContext = new SparkJobContext(sparkContext, confXmlPath, analysisJobXmlPath,
+                propertiesPath);
         final SparkAnalysisRunner sparkAnalysisRunner = new SparkAnalysisRunner(sparkContext, sparkJobContext);
         try {
-            sparkAnalysisRunner.run();
+            final AnalysisResultFuture result = sparkAnalysisRunner.run();
+            if (result.isDone()) {
+                final Map<ComponentJob, AnalyzerResult> resultMap = result.getResultMap();
+                final SimpleAnalysisResult simpleAnalysisResult = new SimpleAnalysisResult(resultMap,
+                        result.getCreationDate());
+                final String resultJobFilePath = getResultJobFilePath(sparkContext, sparkJobContext);
+                if (resultJobFilePath != null) {
+                    final HdfsResource hdfsResource = new HdfsResource(resultJobFilePath);
+                    final OutputStream out = hdfsResource.write();
+                    try {
+                        SerializationUtils.serialize(simpleAnalysisResult, out);
+                    } catch (SerializationException e) {
+                        throw e;
+                    } finally {
+                        FileHelper.safeClose(out);
+                    }
+                }
+            }
         } finally {
             sparkContext.stop();
         }
     }
+
+    private static String getResultJobFilePath(final JavaSparkContext sparkContext, final SparkJobContext sparkJobContext) {
+        String resultPath = sparkJobContext.getResultPath();
```

Comment:
For absolute, but host-less paths (i.e. ```/path/to/file```), it would be good because the path should more or less always be relative to the current cluster's namenode. A path with a full host of another namenode would usually be an error (unless the cluster is using federation).

For "true" relative paths (i.e. ```path/to/file```), it could be a good way to ensure it is still within the default datacleaner folder.